### PR TITLE
Use sklearn's logistic regression for linear probing

### PR DESCRIPTION
**Pull request by ziw-liu**
Optimizing the logistic regression model in sklearn (max-entropy) is the exact same function and objective as optimizing a linear layer with cross-entropy-with-logit loss in torch. The only differences are least-squares vs SGD, and weight regularization. Therefore, by copying the weights and bias from a sklearn model to a torch linear layer, we can use the same deterministic optimization for evaluating classification performance and model interpretation methods that require gradients.

I also include a test case to verify the above: the logits produced by the sklearn and torch models are almost equal (with relative tolerance 0.001).

### CODE REVIEW

**Comment by ziw-liu** [comment]
Example output of `applications/contrastive_phenotyping/evaluation/linear_probing.py`:

```
Trained logistic regression classifier.
Training set accuracy:
              precision    recall  f1-score   support

           0      0.997     0.999     0.998      3836
           1      0.998     0.994     0.996      2076

    accuracy                          0.997      5912
   macro avg      0.997     0.997     0.997      5912
weighted avg      0.997     0.997     0.997      5912

Test set accuracy:
              precision    recall  f1-score   support

           0      0.963     0.987     0.975      4251
           1      0.977     0.938     0.957      2551

    accuracy                          0.968      6802
   macro avg      0.970     0.962     0.966      6802
weighted avg      0.968     0.968     0.968      6802
```
